src/device/usbd.c

```diff
@@ -83,6 +83,7 @@ typedef struct {
   bool (* control_complete ) (uint8_t rhport, tusb_control_request_t const * request);
   bool (* xfer_cb          ) (uint8_t rhport, uint8_t ep_addr, xfer_result_t event, uint32_t xferred_bytes);
   void (* sof              ) (uint8_t rhport);
+  uint8_t (* get_alt_setting ) (uint8_t rhport, tusb_control_request_t const * request); //< (optional) Returns active alternate setting of an opened interface.
```

Comment:
I have a better idea for getting Alt interface, we will define an common usbd_interface_t which contains
- itf_num
- itf_alt
- ep_num
- ep[]

each interface will start with this first, I have been wanting to do this (but being too busy or too lazy), that's why you see all class driver itf start with these. That will also allow to remove the itf2drv and ep2drv mapping. Let's me make an PR for that. This will help with adding Alt Interface.

Reply:
At the moment, I don't understand what you're suggesting, but I'll await the PR, and I'm sure all will become clear.

Reply:
yeah, wait for it ...